src/ShellClients/HideTracker.vala

```diff
@@ -236,9 +276,75 @@ public class Gala.HideTracker : Object {
 
         if (delta_y < 0) { // Only allow swipes upwards
             panel.window.focus (pan_action.get_last_event (0).get_time ());
-            show ();
+            trigger_show ();
         }
 
         return false;
     }
+
+    private void destroy_barrier () {
```

Comment:
I'm not sure haven't looked too much at it and can't test since I don't have a lot of time currently but this might need a destroy_barrier () call in the destructor

Reply:
Used prints to test if there is a memory leak and as far as I can see there is none, Gala.Barrier's destructor gets called by itself and Meta.Barrier gets destroyed.

Reply:
Ahh right I forgot we are using Gala Barrier and not Meta Barrier here :)